src/main/java/com/couchbase/lite/Database.java

```diff
@@ -251,7 +251,9 @@ public void addChangeListener(ChangeListener listener) {
      */
     @InterfaceAudience.Public
     public void compact() throws CouchbaseLiteException {
-        store.compact();
+        synchronized (store) {
+            store.compact();
+        }
```

Comment:
There is nothing wrong with this synchronized. I checked the Database.java code and found that some code path does synchronized (store) but some not. If the intension to make Database thread safe, all of those `synchronized (store)` and `not synchronized (store)` should be revised.